nearuplib/nodelib.py

```diff
@@ -325,7 +355,20 @@ def run_binary(path, home, action, *, verbose=None, shards=None, validators=None
     if output:
         output = open(f'{output}.log', 'w')
 
-    return Popen(command, stderr=output, stdout=output)
+    near = Popen(command, stderr=output, stdout=output)
+    if watch:
+        run_watcher(watch, 'nodocker')
+    return near
+
+
+def run_watcher(watch, docker):
+    watch_script = os.path.abspath(os.path.join(
+        os.path.dirname(__file__), '..', 'watcher.py'))
+    watch_log = open(os.path.expanduser('~/.nearup/logs/watcher.log'), 'w')
+    p = Popen(['python3', watch_script, watch['net'],
+               watch['home'], docker, *watch['args']], stdout=watch_log, stderr=watch_log)
```

Comment:
I had a question here. if don't set stdout=watch_log and stderr=watch_log, all output of watcher goes to current terminal. I can see output every minute of `Our genesis version up to date` `Our binary/docker version up to date`. But once set `stdout=watch_log and stderr=watch_log` only part of output of watcher.py goes to the log file. Why?
(In detail, watcher.py print "Our genesis version up to date" and also do nearup stop, nearup ..., only part of log generated from nearup... called by watcher.py is log to log file, all print simply not print anything)

Reply:
That is due to the buffering. When you run application with terminal it inherits line buffering, but redirecting it to a steam, it prefers huge buffers to avoid frequent writes. To force it to write, you can modify the app to do stdout flush(), or trick the app that you run it in TTY (read about PTY, there are modules for python)

Reply:
Thanks! I see. standard library pty is cumbersome to use, this lib is great: https://ptyprocess.readthedocs.io/en/latest/ but currently we don't have a mechanism to use any pypi lib in nearup. I'll try override print with builitin.print(...flush=True).